### PR TITLE
always create [Library],maximize_library

### PR DESCRIPTION
**Pull request by ronso0**
https://bugs.launchpad.net/mixxx/+bug/1941027

### CODE REVIEW

File: src/util/defs.h [hunk 1]
```diff
@@ -1,7 +1,11 @@
...
 
+#include "preferences/configobject.h"
+
 // Maximum buffer length to each EngineObject::process call.
 //TODO: Replace this with mixxx::AudioParameters::bufferSize()
 constexpr unsigned int MAX_BUFFER_LEN = 160000;
 
 constexpr int kMaxNumberOfDecks = 4;
+
+const ConfigKey kMaximizeLibraryConfigKey = ConfigKey("[Library]", "maximize_library");
```
Comment (Holzhaus) [inline src/util/defs.h:11]: Let's reduce redundancy if we rename it anyway. 

Also, the active verb in `maximize_library` makes it sound like a button (e.g. `[ChannelN],hotcue_X_activate`), which means that you have to press [1.0] and release [0.0] to enable, the press and release again to disable.

But in this case it's actually a flag (1.0 = enabled, 0.0 = disabled). So I think we should use a passive verb here do indicate that (like we do in `[ChannelN],loop_enabled`). 

```suggestion
const ConfigKey kMaximizeLibraryConfigKey = ConfigKey("[Library]", "maximized");
```

Comment (ronso0) [reply src/util/defs.h:11]: I see what you mean, but I think for consistency we should either name it  like `show_previewdeck` and other visibility controls -- or rename all.
Btw despite the redundancy the benefit of `maximize_library` is that the name of the control matches the tooltip which is convenient for skin development.

And actually the reason for using the `[Library]` is solely that it's listed in the respective chapter in the manual, see the bug report. `[Skin],maxmize_library` would be more appropriate IMO.
On that topic: I think we should rename the chapters to `Deck controls`, `Skin controls` etc., much like the submenus in the control picker menu. Also, the `Group` elements in the chapters do decrease the readability IMHO, so I'd appreciate if we'd rename them.

Comment (ronso0) [reply src/util/defs.h:11]: > rename the chapters to `Deck controls`, `Skin controls` etc., much like the submenus in the control picker menu.

image
vs.
image

**Comment by ronso0** [comment]
Btw this approach is a regression right now because the menu action is enabled regardless if the skin actually supplies that control.

**Comment by ronso0** [comment]
@Holzhaus Thanks for your objection, it made me realize this approach would somehow solve the bug but the root cause is elsewhere.

**Comment by ronso0** [comment]
Closing this because I'm not keen on fixing/extending VisibilityConnection and the skin/qml parser to fix the regression.

**Comment by Be-ing** [comment]
> Closing this because I'm not keen on fixing/extending VisibilityConnection and the skin/qml parser to fix the regression.

If integration with the menu bar is causing problems, let's just remove the menu bar entry. The menu bar's days are limited anyway.

**Comment by ronso0** [comment]
> If not, I will.

Go ahead!

**Comment by ronso0** [comment]
Since actually _all_ visibility controls created by (official) skins are affected we should consider a different approach.
Let's discuss this on [Zulip [PreviewDeck], show_previewdeck](https://mixxx.zulipchat.com/#narrow/stream/113295-controller-mapping/topic/.5BPreviewDeck.5D.2C.20show_previewdeck)

**Comment by ronso0** [comment]
Superseeded by #4647

**Pull request closed without merging**